app/assets/stylesheets/petitions/views/shared/_share-petition.scss

```diff
@@ -35,4 +35,9 @@
     margin-top: $icon-size / 2 * -1;
     background-position: center left;
   }
+  .petition-share-whatsapp a {
+    @include media(desktop) {
```

Comment:
Looks good. In future, we may change this to a `@extend %hide-on-desktop` approach if similar requirements come up elsewhere.